lib/manageiq/api/client/mixins/custom_inspect_mixin.rb

```diff
@@ -0,0 +1,24 @@
+module CustomInspectMixin
+  extend ActiveSupport::Concern
+
+  def inspect
+    res = "#{Kernel.instance_method(:to_s).bind(self).call.chomp!('>')} "
+    attrs = instance_variables - custom_inspect_exclusions
+    res << attrs.map { |attr| "#{attr}=#{custom_inspect_value(attr)}" }.join(", ") << ">"
+  end
+
+  private
+
+  def custom_inspect_exclusions
+    return [] unless self.class.const_defined?(:CUSTOM_INSPECT_EXCLUSIONS)
+    self.class::CUSTOM_INSPECT_EXCLUSIONS.map { |attr| "@#{attr}".to_sym }
+  end
+
+  def custom_inspect_value(attr)
+    value = instance_variable_get(attr)
+    return "nil" if value.nil?
+    return "\"#{value}\"" if value.kind_of?(String)
+    return value if value.kind_of?(Hash) || value.kind_of?(Array)
+    "#<#{value.class.name}:#{value.object_id} ...>"
```

Comment:
The object_id here is incorrect.  Prefer .inspect (unless you are intentionally trying to hide the details of an object, but I'm not sure why you'd want to.  Prefer just calling `.inspect`


Given all that, I don't see the need for this custom_inspect_value method.